src/amp-url-creator.js

```diff
@@ -58,25 +58,34 @@ const MAX_DOMAIN_LABEL_LENGTH_ = 63;
  * @param {string} url The complete publisher url.
  * @param {object} initParams Params containing origin, etc.
  * @param {string} opt_cacheUrlAuthority
- * @param {string} opt_viewer_js_version
- * @return {string} the Cache Url.
+ * @param {string} opt_viewerJsVersion
+ * @return {!Promise}
  * @private
  */
 export function constructViewerCacheUrl(url, initParams,
-  opt_cacheUrlAuthority, opt_viewer_js_version) {
+  opt_cacheUrlAuthority, opt_viewerJsVersion) {
   const parsedUrl = parseUrl(url);
-  const cacheDomain = constructCacheDomainUrl_(url, opt_cacheUrlAuthority);
   const protocolStr = parsedUrl.protocol == 'https:' ? 's/' : '';
-  const viewerJsVersion = opt_viewer_js_version ? opt_viewer_js_version :
+  const viewerJsVersion = opt_viewerJsVersion ? opt_viewerJsVersion :
     DEFAULT_VIEWER_JS_VERSION_;
+  const path = parsedUrl.pathname.slice(-1) != '/' ? parsedUrl.pathname + '/' :
+    parsedUrl.pathname;
```

Comment:
is this required by the spec? It seems potentially wrong. /index.html -> /index.html/